### PR TITLE
jwt: add updateKeys method and call in constructor

### LINKED ISSUE #225: JWT middleware needs to get initial JWT set
(Untested, from code only)

The JWT middleware works with an empty set of keys until it runs the update logic in the goroutine for the first time. We should probably just attempt to update in the constructor, before starting the goroutine.

### PR DESCRIPTION
**Pull request by jzelinskie**
Fixes #225.

### CODE REVIEW

File: middleware/jwt/jwt.go [hunk 1]
```diff
@@ -56,44 +56,49 @@ func NewHook(cfg Config) middleware.Hook {
...
-				h.publicKeys = keys
+				h.updateKeys()
 			}
 		}
 	}()
 
 	return h
 }
 
+func (h *hook) updateKeys() {
```
Comment (mrd0ll4r) [inline middleware/jwt/jwt.go]: Make this return an `error` - I believe we want to know if it fails (at least for the initial update).

Comment (jzelinskie) [inline middleware/jwt/jwt.go]: fair point

File: cmd/chihaya/config.go [hunk 2]
```diff
@@ -78,7 +78,11 @@ func (cfg ConfigFile) CreateHooks() (preHooks, postHooks []middleware.Hook, err
 			if err != nil {
 				return nil, nil, errors.New("invalid JWT middleware config: " + err.Error())
 			}
-			preHooks = append(preHooks, jwt.NewHook(jwtCfg))
+			hook, err := jwt.NewHook(jwtCfg)
+			if err != nil {
+				return nil, nil, errors.New("failed to start JWT middleware server: " + err.Error())
```
Comment (mrd0ll4r) [inline cmd/chihaya/config.go]: I wouldn't call this a server, it's not serving anything.

Comment (jzelinskie) [inline cmd/chihaya/config.go]: What should we call it then? It sits in the background updating the JWK set.

Comment (mrd0ll4r) [inline cmd/chihaya/config.go]: I wouldn't assume anything about what failed during the attempt to construct the JWT middleware, I'd just say "failed to start JWT middleware" or something like that

**Comment by jzelinskie** [comment]
ptal

**Pull request merged**